### DISCUSSION TITLE
Is it supported to conditionally turn fetch on/off on different renders/state updates?

### DISCUSSION THREAD

**Discussion by bscaspar** [vercel/swr#685] [Q&A] [answered]
I am using Next and have a statically generated page that checks if the current user is logged in after it loads. I would like to use useSWR to fetch updated data on this page only if the current user is logged in. The check to determine whether they are logged in takes about a half second.

Once the state is updated with the logged in status, is it supported to change the conditional in useSWR? Something like this?
```js
const [session, loading] = useSession();
const { data } = useSWR(!loading && session ? '/api/data' : null, fetcher)
```

**Comment by sergiodxa** [comment, marked as answer]
Yes, that should work without issues
